src/shared/proto/robot_component_status.proto

```diff
@@ -0,0 +1,125 @@
+syntax = "proto3";
+
+/* Data about the status of the break beam */
+message BreakBeamStatus
+{
+    // Whether or not the ball is interrupting the robot’s laser beam.
+    bool ball_in_beam = 1;
+
+    // The reading of the robot’s laser sensor, normalized to [0,1]
+    float break_beam_reading = 2;
+}
+
+/* Data about the status of the chipper and kicker */
+message ChipperKickerStatus
+{
+    // Indicates the time elapsed since chipper was last fired in ms
+    // UINT64_MAX indicates chipper was never fired
+    uint64 time_since_chipper_fired = 1;
+
+    // Indicates the time elapsed since kicker was last fired in ms
+    // UINT64_MAX indicates kicker was never fired
+    uint64 time_since_kicker_fired = 2;
+}
+
+/* Data about the status of the dribbler */
+message DribblerStatus
+{
+    // NOTE: positive means clockwise and negative means counterclockwise
+    int32 dribbler_rpm = 1;
+}
+
+/* Error codes are to be displayed in the visualizer */
+enum ErrorCode
+{
+    NO_ERROR                = 0;  // default message
+    LOW_CAP_MESSAGE         = 1;
+    LOW_BATTERY_MESSAGE     = 2;
```

Comment:
nit: these 3 messages are level-triggered so can be in that section

Reply:
done